mdofail006: cyclic evaluation in fixIO
HasCallStack backtrace:
CallStack (from HasCallStack):
  collectBacktraces, called at libraries/ghc-internal/src/GHC/Internal/Exception.hs:89:13 in ghc-internal:GHC.Internal.Exception
  toExceptionWithBacktrace, called at libraries/ghc-internal/src/GHC/Internal/IO.hs:260:11 in ghc-internal:GHC.Internal.IO
  throwIO, called at libraries/ghc-internal/src/GHC/Internal/System/IO.hs:426:37 in ghc-internal:GHC.Internal.System.IO


